include/revng/Pipeline/Container.h

```diff
@@ -102,26 +108,49 @@ class ContainerBase {
   /// not after(this)->enumerate().contains(Targets);
   ///
   /// returns false if nothing was removed
-  virtual bool remove(const TargetsList &Targets) = 0;
+  virtual bool removeImpl(const TargetsList &Targets) = 0;
+
+  bool remove(const TargetsList &Targets) {
+    bool Result = removeImpl(Targets);
+    Dirty = true;
```

Comment:
I'd set Dirty at the entry and call it a day. No intermediate vars.

No biggie though.

Reply:
Reworked